tensorflow_addons/layers/netvlad.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""NetVLAD keras layer."""
+
+import math
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class NetVLAD(tf.keras.layers.Layer):
+    """Applies NetVLAD to the input.
+
+        "NetVLAD: CNN architecture for weakly supervised place recognition"
+        Relja Arandjelovic, Petr Gronat, Akihiko Torii, Tomas Pajdla, Josef Sivic.
+        https://arxiv.org/abs/1511.07247
+
+    Arguments:
+        num_clusters: The number of clusters to use.
+    Input shape:
+        3D tensor with shape: `(batch_size, time, feature_dim)`.
+    Output shape:
+        2D tensor with shape: `(batch_size, feature_dim * num_clusters)`.
+    """
+
+    def __init__(self, num_clusters, **kwargs):
```

Comment:
Please add typing information and decorate with `@typeguard`:
https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#about-type-hints